lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,199 @@
 """
 Crawler implementation
 """
+
+from pathlib import Path
 from typing import Pattern, Union
 
+import re
+import json
+from bs4 import BeautifulSoup
+import requests
+from requests import HTTPError
+
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import ASSETS_PATH, CRAWLER_CONFIG_PATH
+
+
+class IncorrectSeedURLError(Exception):
+    pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    pass
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    pass
+
+
+class IncorrectHeadersError(Exception):
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    pass
+
 
 class Config:
     """
     Unpacks and validates configurations
     """
 
+    seed_urls: list[str]
```

Comment:
why do you add all these attributes here?

Reply:
were they? https://github.com/fipl-hse/2022-2-level-ctlr/blob/main/lab_5_scrapper/scrapper.py#L7 I do not see